ppl/cmd/zident/user/user.go

```diff
@@ -0,0 +1,232 @@
+package user
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"flag"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/brimsec/zq/ppl/cmd/zident/root"
+	"github.com/go-resty/resty/v2"
+	"github.com/mccanne/charm"
+	"github.com/segmentio/ksuid"
+	"go.uber.org/multierr"
+	"gopkg.in/auth0.v5/management"
+)
+
+const auth0DefaultConnection = "Username-Password-Authentication"
+
+type auth0ClientConfig struct {
+	clientId     string
+	clientSecret string
+	connection   string
+	domain       url.URL
+}
+
+func (c *auth0ClientConfig) FromEnv() error {
+	var err error
+	if c.clientId = os.Getenv("AUTH0_CLIENT_ID"); c.clientId == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_ID not set"))
+	}
+	if c.clientSecret = os.Getenv("AUTH0_CLIENT_SECRET"); c.clientSecret == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CLIENT_SECRET not set"))
+	}
+	if c.connection = os.Getenv("AUTH0_CONNECTION"); c.connection == "" {
+		err = multierr.Append(err, errors.New("AUTH0_CONNECTION not set"))
+	}
+	domain := os.Getenv("AUTH0_DOMAIN")
+	if domain == "" {
```

Comment:
I assume that Auth0 recommends the use of these env vars, right? So we will include the env vars in the deployment.yaml with references to secrets. This is all good and expected, but now our distributed configuration will include a mix of env vars and command line flags.